### PR TITLE
[COTCO-804][PH] Add new crossing endpoint with updated docs, models and tests

### PR DESCRIPTION
**Pull request by peter-hazell**

### CODE REVIEW

File: app/uk/gov/hmrc/transitmovementapi/models/types/ModelTypes.scala [hunk 1]
```diff
@@ -18,30 +18,40 @@ package uk.gov.hmrc.transitmovementapi.models.types
...
-  type Carrier                 = Refined[String, ValidCarrier]
-  type MrnCaptureMethod        = Refined[String, ValidMRNCaptureMethod]
+  private type ValidMrnCaptureMethod =
+    MatchesRegex[W.`"(Scan)|(Manual)"`.T]
+  private type ValidMovementReferenceNumber =
+    MatchesRegex[W.`"""\\d{2}[a-zA-Z]{2}[a-zA-Z0-9]{14}"""`.T]
+  private type ValidTransitUnitType =
+    MatchesRegex[W.`"(Vehicle)|(Container)|(Trailer)"`.T]
+  private type ValidTransitUnitReferenceNumber =
+    MatchesRegex[W.`"""^[\\-A-Z0-9 ]{1,20}$"""`.T]
```
Comment (Flexicoder) [inline app/uk/gov/hmrc/transitmovementapi/models/types/ModelTypes.scala]: thought this was being upped to 50?

File: resources/public/api/conf/1.0/application.raml [hunk 2]
```diff
@@ -36,3 +36,17 @@ uses:
               example: !include examples/submit-transits-request-example.json
           responses:
            200:
+      /crossing:
+        post:
+          displayName: Submit details of a crossing
+          description: This endpoint takes a JSON body that contains details of a crossing including any associated vehicles, trailers or containers along with the movement reference numbers of transit movements that are inside them.
```
Comment (Flexicoder) [inline resources/public/api/conf/1.0/application.raml]: Is it worth adding something like "MRN's that are known to be CTC related", i.e. so its clear we are not expecting everything?

File: resources/public/api/conf/1.0/schemas/submit-crossing-request.json [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+              "Container",
+              "Trailer",
+              "Vehicle"
+            ]
+          },
+          "referenceNumber": {
+            "type": "string",
+            "description": "The reference number of the transit unit that the transit movement is in",
+            "minLength": 1,
+            "maxLength": 20,
```
Comment (Flexicoder) [inline resources/public/api/conf/1.0/schemas/submit-crossing-request.json]: Again thought this was 50?

File: resources/public/api/conf/1.0/schemas/submit-transits-request.json [hunk 4]
```diff
@@ -56,57 +58,52 @@
...
+        "Vehicle",
+        "Container",
+        "Trailer"
+      ]
+    },
+    "transitUnitReferenceNumber": {
+      "type": "string",
+      "description": "The reference number of the transit unit that the transit movement is on",
+      "minLength": 1,
+      "maxLength": 20,
```
Comment (Flexicoder) [inline resources/public/api/conf/1.0/schemas/submit-transits-request.json]: 50?

File: test-common/uk/gov/hmrc/transitmovementapi/helpers/CrossingGenerator.scala [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+      transitUnitType            <- transitUnitTypeGenerator
+      transitUnitReferenceNumber <- transitUnitReferenceNumberGenerator
+    } yield TransitUnit(transitUnitType, transitUnitReferenceNumber, movementReferenceNumbers)
+
+  private def transitUnitTypeGenerator: Gen[TransitUnitType] =
+    Gen.oneOf[TransitUnitType]("Container", "Trailer", "Vehicle")
+
+  private def transitUnitReferenceNumberGenerator: Gen[TransitUnitReferenceNumber] =
+    RegexpGen
+      .from("""^[\-A-Z0-9 ]{1,20}$""")
```
Comment (Flexicoder) [inline test-common/uk/gov/hmrc/transitmovementapi/helpers/CrossingGenerator.scala]: 50?

File: test-common/uk/gov/hmrc/transitmovementapi/helpers/TransitGenerator.scala [hunk 6]
```diff
@@ -17,72 +17,57 @@
...
-    Gen.option(RegexpGen.from("""^[\-A-Z0-9 ]{1,16}$""").map(vrn => Json.toJson(vrn).as[VehicleReferenceNumber]))
+  private def transitUnitTypeGenerator: Gen[TransitUnitType] =
+    Gen.oneOf[TransitUnitType]("Container", "Trailer", "Vehicle")
 
-  private def crossingIdGenerator(withDefaultCrossingId: Option[String]): Gen[String] = {
-    Gen.const(withDefaultCrossingId.fold(UUID.randomUUID().toString)(id => id))
-  }
+  private def transitUnitReferenceNumberGenerator: Gen[TransitUnitReferenceNumber] =
+    RegexpGen
+      .from("""^[\-A-Z0-9 ]{1,20}$""")
```
Comment (Flexicoder) [inline test-common/uk/gov/hmrc/transitmovementapi/helpers/TransitGenerator.scala]: 50?

File: resources/public/api/conf/1.0/application.raml (hunk 2 above)
Comment (peter-hazell) [reply resources/public/api/conf/1.0/application.raml]: good point

**Pull request merged**